tools/libraries/src/Server.ts

```diff
@@ -220,21 +239,38 @@ export class Server extends TypedEmitter<ServerEvents> {
       senderToken,
       senderWallet,
     }
+    if (minExpiry) {
+      params.minExpiry = minExpiry
+    }
     if (proxyingFor) {
       params.proxyingFor = proxyingFor
     }
     return this.callRPCMethod<OrderERC20>(
       'getSenderSideOrderERC20',
       params
     ).then((order) => {
+      const errors = this.compare(params, order)
+      if (errors.length) {
+        throw new Error(
+          `Server response differs from request params: ${errors}`
+        )
+      } else if (!isValidOrderERC20(order)) {
+        throw new Error(
+          `Server responded with invalid order: ${JSON.stringify(order)}`
+        )
+      }
       return orderERC20PropsToStrings(order)
     })
   }
 
   public async getPricingERC20(
-    pairs: { baseToken: string; quoteToken: string }[]
+    pairs: { baseToken: string; quoteToken: string }[],
+    minExpiry?: string
   ): Promise<Pricing[]> {
-    return this.callRPCMethod<Pricing[]>('getPricingERC20', [pairs])
+    return this.callRPCMethod<Pricing[]>('getPricingERC20', {
```

Comment:
Should we also check the fields in the body here ?

Reply:
Fields are checked on the `get*OrderERC20` calls because you want the response params to match the request params. Since the `getPricingERC20` response is more complex (with lists of pairs and lists of levels) I say we just let it go through to the client and they can work it out.

Reply:
Okay !